systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java

```diff
@@ -100,20 +109,120 @@ void testRackAware() {
+        final String defaultKafkaClientsPodName =
+                ResourceManager.kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+                .withUsingPodName(defaultKafkaClientsPodName)
+                .withTopicName(TOPIC_NAME)
+                .withNamespaceName(NAMESPACE)
+                .withClusterName(CLUSTER_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withListenerName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+                .build();
+
+        internalKafkaClient.verifyProducedAndConsumedMessages(
+                internalKafkaClient.sendMessagesPlain(),
+                internalKafkaClient.receiveMessagesPlain()
         );
     }
 
+    @Test
+    @Tag(CONNECT)
+    @Tag(REGRESSION)
+    void testRackAwareConnect() throws Exception {
+        installClusterOperator(NAMESPACE, CO_OPERATION_TIMEOUT_SHORT);
+
+        String wrongRackKey = "wrong-key";
+        String rackKey = "rack-key";
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewRack()
+                            .withTopologyKey(rackKey)
+                        .endRack()
+                        .addToConfig("replica.selector.class", "org.apache.kafka.common.replica.RackAwareReplicaSelector")
+                    .endKafka()
+                .endSpec().done();
+
+        KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+        String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+        LOGGER.info("Deploy KafkaConnect with wrong rack-aware topology key: {}", wrongRackKey);
+        KafkaConnectResource.kafkaConnectWithoutWait(KafkaConnectResource.defaultKafkaConnect(CLUSTER_NAME, CLUSTER_NAME, 1)
+                .editSpec()
+                    .withNewRack()
+                        .withTopologyKey(wrongRackKey)
+                    .endRack()
+                .endSpec().build());
+
+        PodUtils.waitForPendingPod(CLUSTER_NAME + "-connect");
+        List<String> connectWrongPods = kubeClient().listPodNames(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND);
+        String connectWrongPodName = connectWrongPods.get(0);
+        LOGGER.info("Waiting for ClusterOperator to get timeout operation of incorrectly set up KafkaConnect");
+        Thread.sleep(CO_OPERATION_TIMEOUT_SHORT + 10000);
+
+        KafkaConnect kc = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get();
+        Condition kcCondition = kc.getStatus().getConditions().get(0);
+        assertThat("NotReady", is(kcCondition.getType()));
+        assertThat("TimeoutException", is(kcCondition.getReason()));
```

Comment:
Didn't you had some `waitFor` methods which can do this in a more efficient way?